Toolchain/BuildMold.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# This script builds the mold linker that can optionally be used for linking
+# the SerenityOS userland.
+set -e
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+NPROC="nproc"
+SYSTEM_NAME="$(uname -s)"
+
+if [ "$SYSTEM_NAME" = "OpenBSD" ]; then
+    NPROC="sysctl -n hw.ncpuonline"
+elif [ "$SYSTEM_NAME" = "FreeBSD" ]; then
+    NPROC="sysctl -n hw.ncpu"
+elif [ "$SYSTEM_NAME" = "Darwin" ]; then
+    NPROC="sysctl -n hw.ncpu"
+fi
+
+[ -z "$MAKEJOBS" ] && MAKEJOBS=$($NPROC)
+
+mkdir -p "$DIR"/Tarballs
+pushd "$DIR"/Tarballs
+
+[ ! -d mold ] && git clone https://github.com/rui314/mold.git
+
+cd mold
+
+git pull
```

Comment:
We should checkout a specific version, otherwise this is going to be the wild west.
Or better yet, download from a tagged release instead of cloning from git. 
`https://github.com/rui314/mold/releases/tag/v1.0.2`

Reply:
Sure. The reason I opted to do a git clone initially is because 1.0.1 is not able to build our userland, but I guess since there's 1.0.2, we can use that.